web/client/utils/PrintUtils.js

```diff
@@ -229,6 +229,45 @@ export const getMapSize = (layout, maxWidth) => {
 
 export const mapProjectionSelector = (state) => state?.print?.map?.projection ?? "EPSG:3857";
 
+export function parseCreditRemovingTagsOrSymbol(creditText = "") {
+    let parsedCredit = creditText;
+    do {
+        let tagStartIndex = parsedCredit.indexOf("<");
+        let tagEndIndex = parsedCredit.indexOf(">");
+        if (tagStartIndex !== -1 && tagEndIndex !== -1) {
+            parsedCredit = parsedCredit.replace(parsedCredit.substring(tagStartIndex, tagEndIndex + 1), "");
+        }
+    } while (parsedCredit.includes("<") || parsedCredit.includes(">"));
+    let hasOrSymbol = parsedCredit && parsedCredit.includes("|");
+    if (hasOrSymbol) {
+        parsedCredit = parsedCredit?.replaceAll("|", "")?.replaceAll("  ", " ");
+    }
+    return parsedCredit;
+}
+/**
+ * Gets the credits of layers in one text with '|' separated
+ * @param  {object} layers the map layers for print
+ * @returns {string}       the layers credits as a text '|' separated
+ * @memberof utils.PrintUtils
+ */
+export const getLayersCredits = (layers) => {
+    const credits = layers.reduce((cum, lay) => {
+        let cumCredit = cum;
+        let layerCredit = lay?.credits?.title;
+        if (layerCredit) {
+            // remove tag <> and symbols like: |
+            let hasOrSymbol = layerCredit.includes('|');
+            const hasHtmlTag = layerCredit.includes('<');
+            if (hasHtmlTag || hasOrSymbol) {
+                layerCredit = parseCreditRemovingTagsOrSymbol(layerCredit);
+            }
+            cumCredit = cumCredit ? cumCredit + " | " + layerCredit : layerCredit;
+        }
+        return cumCredit;
+    }, '');
+    return credits;
+};
```

Comment:
your version is a bit cumbersome and can be simplified this way
```suggestion
export const getLayersCredits = (layers) => {
    const layerCredits = layers.map((layer) => {
        const layerCreditTitle = layer?.credits?.title || '';
        const hasOrSymbol = layerCreditTitle.includes('|');
        const hasHtmlTag = layerCreditTitle.includes('<');
        const layerCredit = (hasHtmlTag || hasOrSymbol)
            ? parseCreditRemovingTagsOrSymbol(layerCreditTitle)
            : layerCreditTitle;
        return layerCredit;
    }).join(' | ');
    return layerCredits;
};

```

Reply:
Nice way, thanks